File: python/tree.py
"""working....success.."""
class Node:
	def __init__(self,item):
		self.left=None
		self.right=None
		self.val=item
def inorder(root):
	if root:
		inorder(root.left)
		print(str(root.val)+"->",end='')
		inorder(root.right)
def postorder(root):
	if root:
		postorder(root.left)
		postorder(root.right)
		print(str(root.val)+"->",end='')
def preorder(root):
	if root:
		print(str(root.val)+"->",end='')
		preorder(root.left)
		preorder(root.right)

#driver code...
root = Node(1)
root.left=Node(2)
root.right=Node(3)
root.left.left=Node(4)
root.left.right=Node(5)
print("\ninorder traversal")
inorder(root)
print("\npreorder traversal")
preorder(root)
print("\npostorder traversal")
postorder(root)

File: java/Tree.java
import java.util.*;
//working...success...

public class Tree{
	Node root;
	Tree(){root=null;}
	void postorder(Node node){
		if(node==null){return;}
		postorder(node.left);
		postorder(node.right);
		System.out.print(node.item+"->");}
    void inorder(Node node){
			if(node==null){return;}
			inorder(node.left);
			System.out.print(node.item+"->");
			inorder(node.right);}
	void preorder(Node node){
		if(node==null){return;}
		System.out.print(node.item+"->");
		preorder(node.left);
		preorder(node.right);}

	//driver code..
	public static void main(String f[]){
		Tree tree=new Tree();
		tree.root=new Node(1);
		tree.root.left=new Node(12);
		tree.root.right=new Node(9);
		tree.root.left.left=new Node(5);
		tree.root.left.right=new Node(6);

		System.out.println("inorder traversal");
		tree.inorder(tree.root);
		System.out.println("preorder traversal");
		tree.preorder(tree.root);
		System.out.println("postorder traversal");
		tree.postorder(tree.root);
	}
}
class Node{
	int item;
	Node left,right;
	public Node(int key){item=key;left=right=null ;}}